contracts/VotingAdapter.vy

```diff
@@ -0,0 +1,208 @@
+# @version 0.3.7
+
+"""
+@title Voting Adapter
+@author Lido Finance
+@license MIT
+@notice Used to allow voting with tokens under vesting
+"""
+
+from vyper.interfaces import ERC20
+
+
+interface IDelegation:
+    def setDelegate(
+        _id: bytes32,
+        _delegate: address,
+    ): nonpayable
+
+
+interface IVoting:
+    def vote(
+        _voteId: uint256,
+        _supports: bool,
+        _executesIfDecided_deprecated: bool,
+    ): nonpayable
+
+
+event ERC20Recovered:
+    token: address
+    amount: uint256
+
+
+event ETHRecovered:
+    amount: uint256
+
+
+event OwnerChanged:
+    owner: address
+
+
+ZERO_BYTES32: constant(
+    bytes32
+) = 0x0000000000000000000000000000000000000000000000000000000000000000
+
+VOTING_CONTRACT_ADDR: immutable(address)
+SNAPSHOT_DELEGATE_CONTRACT_ADDR: immutable(address)
+DELEGATION_CONTRACT_ADDR: immutable(address)
+
+owner: public(address)
+
+
+@external
+def __init__(
+    voting_addr: address,
+    snapshot_delegate_addr: address,
+    delegation_addr: address,
+    owner: address,
+):
+    """
+    @notice Initialize source contract implementation.
+    @param voting_addr Address of the Voting contract
+    @param snapshot_delegate_addr Address of the Shapshot Delegate contract
+    @param delegation_addr Address of the voting power delegation contract
+    @param owner Address to recover tokens and ether to
+    """
+    assert owner != empty(address), "zero owner"
+    self.owner = owner
+    VOTING_CONTRACT_ADDR = voting_addr
```

Comment:
check that this is not null

Reply:
One can imagine when these contracts will be used by someone outside LIdo. Plus Lido itself who might want to vest non-LDO tokens. In this case voting adapter makes no sense.
We have checks regarding the voting address being not zero address in https://github.com/lidofinance/lido-vesting-escrow/blob/e525826156e6ad0d5d2f2c098d145a61d522c9c6/contracts/VestingEscrow.vy#L279

Reply:
Done